src/handling_of_spatial_data.Rmd

```diff
@@ -0,0 +1,123 @@
+Here, spatial clusters are equal to the number of folds chosen. 
+These spatially disjoint subsets of the data introduce a spatial distance between training and test set. 
+This reduces the influence of spatial autocorrelation and subsequently also the overoptimistic predictive accuracy estimates.
+The example in Figure 1 shows a five-fold nested cross-validation setting and exhibits the difference between spatial and non-spatial partitioning.
+The nested approach is used when hyperparameter tuning is performed.
+
+## How to use spatial partitioning in mlr
+
+Spatial partitioning can be used when performing cross-validation. 
+In any [&resample] call you can choose `SpCV` or `RepSpCV` to use it. 
+There are some prerequisites for this:
+
+When specifying the [task](task.md), you need to explicitly state that the task is `spatial` one by setting `spatial = TRUE`. 
+Next, there need to be two variables named `x` and `y` in your data set that store the coordinate information. 
+If this applies, the coordinates will be used for spatial partitioning if `SpCV` or `RepSpCV` are selected as resampling strategies. 
+
+Also, these variables **will be removed** for any [train()](&train) or [predict()](&predict) call. 
+This means they will not be used as predictors within the specified learner but only to set up the sampling for the cross-validation task.
+
+Coordinates must be named `x` and `y` to be used for spatial partitioning. 
+If named differently and `spatial = TRUE` was set during task creation, an error will occur. 
+
+## Examples
+
+The `basque` data set serves as an example data set for spatial modeling tasks in `mlr`.
+The [task](task.md) attribute `spatial` is set to `TRUE` to indicate that the data set also stores coordinate information.
+
+In this example, the "Random Forest" algorithm (package [ranger](&ranger::ranger)) is used to model a binomial response variable.
+
+For performance assessment, a repeated spatial cross-validation with 5 folds and 10 repetitions is chosen.
+
+### Spatial Cross-Validation
+
+```{r, eval = TRUE}
+data("bc.task.spatial")
+bc.task.spatial
+
+learner.glm = makeLearner("classif.ranger", predict.type = "prob")
+
+resampling = makeResampleDesc("SpRepCV", fold = 5, reps = 10)
+
+set.seed(123)
+out = resample(learner = learner.glm, task = bc.task.spatial,
+  resampling = resampling, measures = list(auc))
+
+mean(out$measures.test$auc)
+```
+
+We can check for the introduced spatial autocorrelation bias here by performing the same modeling task using a non-spatial partitioning setting.
+To force non-spatial cross-validation here, variables `x` and `y` need to be removed from the data set because they are not used for partitioning when using `RepCV` and also should not be used as predictors. 
+
+Additionally, the `spatial` task attribute needs to be set to `FALSE`.
+`mlr` will issue a warning message if it detects variables named `x` or `y` in combination with  `spatial = FALSE`.
```

Comment:
Removed the warning.

Reply:
ah just saw it - what was the reason for that?

Reply:
It's annoying :)

Reply:
Hmm - I never name any variables `x` or `y`  so I thought it would only apply if really somebody forgot to remove the coordinates by accident. 
But yeah, if somebody frequently names variables `x` or `y` then its annoying.

The problem is that if this happens with spatial data without notice, the coordinates are used as predictors (which is usually not desired by the user). 
However, I cannot think of any other solution to this as there is no way of erroring if `x` or `y` is among the variable names when any kind of CV is used.

Do you have any idea how this could be handled differently than by issuing a warning msg?

Reply:
Well, in general we tend to take the view that the user should know what they're doing when it's not obvious that something is wrong. You can do all sorts of stuff that doesn't make sense with mlr without getting a warning.

If the user has spatial data, I think that we can assume that they do their homework and read up on how to handle this in mlr.

Reply:
Ok, fair enough :smile: